auth/server/base/src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -1115,6 +1116,15 @@ private void handleOne(final Callback[] callbacks, final int idx) throws IOExcep
                         addPublicCredential(credential);
                     }
                     handleOne(callbacks, idx + 1);
+                } else if (callback instanceof PrincipalAuthorizeCallback) {
+                    PrincipalAuthorizeCallback authorizeCallback = (PrincipalAuthorizeCallback) callback;
+                    Principal principal = authorizeCallback.getPrincipal();
+                    // always re-set the principal to ensure it hasn't changed.
+                    setAuthenticationPrincipal(principal);
+                    boolean authorized = authorize(principal);
```

Comment:
Since the authentication principal has already been set and since the same principal is being used for the authorization check, I think just `authorize()` could be used here instead.

Reply:
Make sense! Done!